gluten-core/src/main/scala/org/apache/gluten/backendsapi/ListenerApi.scala

```diff
@@ -16,11 +16,11 @@
  */
 package org.apache.gluten.backendsapi
 
-import org.apache.spark.SparkConf
+import org.apache.spark.{SparkConf, SparkContext}
 
 trait ListenerApi {
-  def onDriverStart(conf: SparkConf): Unit = {}
+  def onDriverStart(sc: SparkContext, conf: SparkConf): Unit = {}
   def onDriverShutdown(): Unit = {}
-  def onExecutorStart(conf: SparkConf): Unit = {}
+  def onExecutorStart(executorID: String, conf: SparkConf): Unit = {}
   def onExecutorShutdown(): Unit = {}
```

Comment:
nit

```suggestion
  def onDriverStart(sc: SparkContext, pc: PluginContext): Unit = {}
  def onDriverShutdown(): Unit = {}
  def onExecutorStart(pc: PluginContext): Unit = {}
  def onExecutorShutdown(): Unit = {}
```

to keep up with Spark's plugin APIs

Reply:
addressed